### PR TITLE
Remove the bitfield value from fu_device_register_private_flag()

### PR DESCRIPTION
**Pull request by hughsie**
Although using a bitmask is quicker, we're already looking for the private flag item to check that it was registered. Using a string means we don't need to worry about bit offsets and device hierarchy when defining flags on things we later subclass.

Note, I've also deliberately not done a define for the flag string -- e.g. `#define FU_AVER_HID_FLAG_DUAL_ISP "dual-isp"` for a few reasons:

 * It's already effectively API, as it's used in the quirk file directly
 * We're checking the flag exists every time we check and modify the value
 * We don't have the opportunity to swap the name and value like 1b77e63a

Also, plugin private flags should be documented in the `README.md`, not in code.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-device.c [hunk 1]
```diff
@@ -592,36 +605,21 @@ fu_device_remove_private_flag(FuDevice *self, guint64 flag)
...
+
 	g_return_val_if_fail(FU_IS_DEVICE(self), FALSE);
-#ifndef SUPPORTED_BUILD
-	if (fu_device_private_flag_item_find_by_val(self, flag) == NULL) {
-		g_critical("%s flag 0x%x is unknown -- use fu_device_register_private_flag()",
+	g_return_val_if_fail(flag != NULL, FALSE);
+
+	item = fu_device_private_flag_item_find_by_flag(self, flag);
+	if (item == NULL) {
+		g_critical("%s flag %s is unknown -- use fu_device_register_private_flag()",
```
Comment (superm1) [inline libfwupdplugin/fu-device.c:617]: this changes the behavior to show the message even on supported builds.  Is that really what we want?

**Review by superm1 — APPROVED** [review]
One question, but otherwise I'm happy with this change

File: libfwupdplugin/fu-device.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-device.c:617]: Maybe not; I'll stick the guards back in.

**Comment by hughsie** [comment]
@superm1 the other thing we could do is merge the FuDeviceInternalFlags with this change? I mean the distinction between a libfwupd FwupdDeviceFlag and the one used internally by the daemon is clear, but what's the distinction between the private flags and the internal flags? Internal flags are used by more than one plugin? I'm also not sure how to document those if we did that -- on the assumption we'd also remove the hash defines. Ideas welcome.

**Comment by hughsie** [comment]
@superm1 I like https://github.com/fwupd/fwupd/pull/7637 more! :)

**Pull request closed without merging**